lstchain/reco/dl1_to_dl2.py

```diff
@@ -417,3 +429,85 @@ def apply_models(dl1, classifier, reg_energy, reg_disp_vector, custom_config={})
     dl2['gammaness'] = probs
     return dl2
 
+
+
+def get_source_dependent_parameters(data, config={}):
+
+    """Get parameters for source-dependent analysis .
+
+    Parameters:
+    -----------
+    data: Pandas DataFrame
+    config: dictionnary containing configuration
+    
+    """
+
+    src_dep_params = pd.DataFrame(index=data.index)
+
+    is_simu = 'mc_type' in data.columns
+    
+    if is_simu:
+        if (data['mc_type'] == 0).all():
+            data_type = 'mc_gamma'
+        else:
+            data_type = 'mc_proton'
+    else:
+        data_type = 'real_data'
+    
+    expected_src_pos_x_m, expected_src_pos_y_m = get_expected_source_pos(data, data_type, config)
+
+    src_dep_params['expected_src_x'] = expected_src_pos_x_m
+    src_dep_params['expected_src_y'] = expected_src_pos_y_m
+    
+    src_dep_params['dist'] = np.sqrt((data['x'] - expected_src_pos_x_m)**2 + (data['y'] - expected_src_pos_y_m)**2)
```

Comment:
What is x here? Shouldn't be the relevant sign be the sign of the source in the shower coordinates? As in `ctapipe.image.hillas.pixel_to_shower_coordinates`?

Reply:
I.e. `np.sign(disp)` from below.

Reply:
`x` is x-position of c.o.g in camera coordinate, so `dist` is the distance between cog and expected source position also in camera coordinate.

About the `time_gradient` and `skewness`, the sign of those parameters is already defined in shower coordinate. At that time, its sign depends on `psi` angle. Here the sign is corrected according to the `psi` (the relation between the position of cog and expected source)

Reply:
`np.abs(skewness) * np.sign(disp)` is also one of the option

Reply:
Are you sure taking the x camera position works for all possible source location? I didn't think it through.

Reply:
You're right... I misunderstood...
Let me modify the codes.

Reply:
done!